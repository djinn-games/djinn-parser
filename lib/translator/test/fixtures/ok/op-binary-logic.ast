{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "logic_operators",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 3,
            "dataType": "bool",
            "value": true
          },
          "right": {
            "type": "Literal",
            "line": 3,
            "dataType": "bool",
            "value": false
          },
          "operator": "and",
          "line": 3
        },
        "line": 3
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 4,
            "dataType": "bool",
            "value": false
          },
          "right": {
            "type": "OperationExpression",
            "left": {
              "type": "Literal",
              "line": 4,
              "dataType": "int",
              "value": 5
            },
            "right": {
              "type": "Literal",
              "line": 4,
              "dataType": "int",
              "value": 10
            },
            "operator": "lt",
            "line": 4
          },
          "operator": "or",
          "line": 4
        },
        "line": 4
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 5,
            "dataType": "bool",
            "value": true
          },
          "right": {
            "type": "OperationExpression",
            "right": {
              "type": "Literal",
              "line": 5,
              "dataType": "bool",
              "value": true
            },
            "operator": "not",
            "line": 5
          },
          "operator": "and",
          "line": 5
        },
        "line": 5
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 6,
            "dataType": "bool",
            "value": true
          },
          "right": {
            "type": "OperationExpression",
            "right": {
              "type": "Literal",
              "line": 6,
              "dataType": "bool",
              "value": true
            },
            "operator": "not",
            "line": 6
          },
          "operator": "or",
          "line": 6
        },
        "line": 6
      }
    ],
    "line": 2
  },
  "line": 1
}