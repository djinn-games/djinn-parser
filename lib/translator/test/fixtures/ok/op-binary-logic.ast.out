{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "LogicalExpression",
        "left": {
          "type": "Literal",
          "value": true,
          "raw": "true"
        },
        "operator": "&&",
        "right": {
          "type": "Literal",
          "value": false,
          "raw": "false"
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "LogicalExpression",
        "left": {
          "type": "Literal",
          "value": false,
          "raw": "false"
        },
        "operator": "||",
        "right": {
          "type": "BinaryExpression",
          "left": {
            "type": "Literal",
            "value": 5,
            "raw": "5"
          },
          "operator": "<",
          "right": {
            "type": "Literal",
            "value": 10,
            "raw": "10"
          }
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "LogicalExpression",
        "left": {
          "type": "Literal",
          "value": true,
          "raw": "true"
        },
        "operator": "&&",
        "right": {
          "type": "UnaryExpression",
          "operator": "!",
          "prefix": true,
          "argument": {
            "type": "Literal",
            "value": true,
            "raw": "true"
          }
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "LogicalExpression",
        "left": {
          "type": "Literal",
          "value": true,
          "raw": "true"
        },
        "operator": "||",
        "right": {
          "type": "UnaryExpression",
          "operator": "!",
          "prefix": true,
          "argument": {
            "type": "Literal",
            "value": true,
            "raw": "true"
          }
        }
      }
    }
  ]
}
